mParticle-Apple-SDK/Next Gen/MPEventLogging.h

```diff
@@ -0,0 +1,50 @@
+//
+//  MPEventLogging.h
+//  mParticle-Apple-SDK
+//
+//  Created by Ben Baron on 6/29/23.
+//
+
+#import <Foundation/Foundation.h>
+#import "MPComponent.h"
+
+@class MPEvent;
+@class MPBaseEvent;
+@class MPCommerceEvent;
+@class MPNetworkPerformance;
+
+typedef NS_ENUM(NSUInteger, MPExecStatus) {
+    MPExecStatusSuccess = 0,
+    MPExecStatusFail,
+    MPExecStatusMissingParam,
+    MPExecStatusDisabledRemotely,
+    MPExecStatusEnabledRemotely,
+    MPExecStatusOptOut,
+    MPExecStatusDataBeingFetched,
+    MPExecStatusInvalidDataType,
+    MPExecStatusDataBeingUploaded,
+    MPExecStatusServerBusy,
+    MPExecStatusItemNotFound,
+    MPExecStatusDisabledInSettings,
+    MPExecStatusNoConnectivity
+};
+
+NS_ASSUME_NONNULL_BEGIN
+
+@interface MPEventLogging : NSObject <MPComponent>
+
+@property (nonatomic, strong, nullable) NSMutableSet<MPEvent *> *eventSet;
+
+- (void)leaveBreadcrumb:(nonnull MPEvent *)event completionHandler:(void (^ _Nonnull)(MPEvent * _Nonnull event, MPExecStatus execStatus))completionHandler;
+- (void)logError:(nullable NSString *)message exception:(nullable NSException *)exception topmostContext:(nullable id)topmostContext eventInfo:(nullable NSDictionary *)eventInfo completionHandler:(void (^ _Nonnull)(NSString * _Nullable message, MPExecStatus execStatus))completionHandler;
```

Comment:
Haven't we assumed that we were going to collapse error and exception (in your case "crash") on a single function?

Reply:
See my previous comment, that change will come later and is outside of scope for this initial work.

Reply:
Great